platforms/ide/tooling-api/src/main/java/org/gradle/tooling/ProblemAwareFailure.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.tooling;
+
+import org.gradle.api.Incubating;
+import org.gradle.tooling.events.problems.ProblemReport;
+
+import java.util.List;
+
+/**
+ * Represents a build failure that has associated problems.
+ *
+ * @since 8.11
+ */
+@Incubating
+public interface ProblemAwareFailure extends Failure {
```

Comment:
🤔 Why do we need a new type here? Why don't we add a `getProblems()` method to `Failure` directly?

Reply:
You are right. I removed this interface.